### ISSUE TITLE
Clearer porcupine math in Chapter 2

### ISSUE THREAD

**Issue by searchivarius** [hal3/ciml#209]
Hi! I just started reading the book to referesh some of my memories. I really like the pedagogical approach.
However, in the porcupine example I couldn't really wrap my head around the math your are doing. I understand the idea and can compute things in 2d, but I bet you would have hard time extending such as a simple approach to an dimensional space (with all the nitty-gritty details explained!). 

Simply speaking, it seems to be quite non-trivial to see how you apply the Pythagorean theoream in an n-d space. So, I suggest an alternative, which may be more clear. In the attached picture, I draw a 2d projection. It is easy to explain how, using the Pythagorean theorem, one could compute the length of the diagonal of the cube: 2 sqrt(d). 

The mere fact that it's growing unboundly is already an anomaly, which explains a lot of things. To compute the diamter of the inner (black circle) you should simply subtract twice the diameter of the two spheres and twice the length of EB (see the picture). To compute the length of EB, you compute OB using a Pythagorean theorem. Again, this would be quite easy an intuitive.

![porcupine math](https://cloud.githubusercontent.com/assets/825650/18536982/08ede636-7ad0-11e6-8b92-fa4ece292066.jpg)

**Comment by searchivarius** [comment]
So, in this comment, I just draw a better projection picture that is valid for d > 2. Dashes indicate invisible lines. So, you can see the diagonal AB is visible, then it is covered by the sphere (dashed line), then it is kinda visible in the center. 

The boundary of the central sphere is also dashed when it's covered by other spheres. You can also easily find the projections of points where the central sphere touches other spheres. 

As dimensionality grows, the cosine of the angle between AB and any adjacent cube edge converges to zero and the diagonal approaches a perpindicular to the projection surface (follows from calculating the scalar product between (1,0,0, ....,0) and (1,1,1...,1)). Again, another high-dimensional anomaly! Consequently, the **projection** of the point where the middle sphere touches the left upper sphere **approaches** point A (in the projection). 

This again, makes a lot of sense, because the diameter of the central sphere (2sqrt(d)-2) converges to the length of the diagonal (2sqrt(d)). So, as dimensionality grows the central sphere grows in diameter and eventually starts to stick out the cube. In that, it remains perfectly round and only touches other spheres. 

To sum up, this is not a porcupine, but a **packing** quirk that we don't observe in 2d and 3d. Or so it seems. Does it make sense?

![porcupine math better picture](https://cloud.githubusercontent.com/assets/825650/18541769/3fa99038-7af5-11e6-8d50-e774ddfac44c.jpg)

**Comment by searchivarius** [comment]
PS: another suggestion. If you talk about anomalies, another one that might be worth mentioning is that nearly all the volume of the sphere is concentrated in a shell near surface. This seems to have a lot of implications.
